### PR TITLE
Fix display of data-function

### PR DESCRIPTION
**Pull request by chmst**
Pull Request for Issue #38958

Open: removing modal from com_sheduler. 

### Summary of Changes
Fix display of data-function in com_modules, remove data-function in com_scheduler
Remove modal view

### Testing Instructions
Compare the html code of Select screens of com_modules and com_scheduler
Try to find a situation where a modal view is missing.

### Actual result BEFORE applying this Pull Request
grafik

### Expected result AFTER applying this Pull Request

No data-function when $function is empty in com_modules, No data-function in com_scheduler
grafik

Remove Modal view

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: administrator/components/com_modules/tmpl/select/default.php [hunk 1]
```diff
@@ -63,7 +63,9 @@ class="form-control" id="comModulesSelectSearch"
                     <?php $link = 'index.php?option=com_modules&task=module.add&client_id=' . $this->state->get('client_id', 0) . $this->modalLink . '&eid=' . $item->extension_id; ?>
                     <?php $name = $this->escape($item->name); ?>
                     <?php $desc = HTMLHelper::_('string.truncate', $this->escape(strip_tags($item->desc)), 200); ?>
-                <a href="<?php echo Route::_($link); ?>" class="new-module mb-3 comModulesSelectCard" data-function="' . $this->escape($function) : ''; ?>" aria-label="<?php echo Text::sprintf('COM_MODULES_SELECT_MODULE', $name); ?>">
+                <a href="<?php echo Route::_($link); ?>" class="new-module mb-3 comModulesSelectCard"
+                    <?php echo !empty($function) ? 'data-function="' . $this->escape($function)  . '"': '' ;?>
```
Comment (Quy) [inline administrator/components/com_modules/tmpl/select/default.php:67]:
```suggestion
                    <?php echo !empty($function) ? 'data-function="' . $this->escape($function)  . '"': ''; ?>
```

Comment (Quy) [inline administrator/components/com_modules/tmpl/select/default.php]:
```suggestion
                    <?php echo !empty($function) ? 'data-function="' . $this->escape($function) . '"': ''; ?>
```

File: administrator/components/com_modules/tmpl/select/default.php [hunk 2]
```diff
@@ -63,7 +63,9 @@ class="form-control" id="comModulesSelectSearch"
                     <?php $link = 'index.php?option=com_modules&task=module.add&client_id=' . $this->state->get('client_id', 0) . $this->modalLink . '&eid=' . $item->extension_id; ?>
                     <?php $name = $this->escape($item->name); ?>
                     <?php $desc = HTMLHelper::_('string.truncate', $this->escape(strip_tags($item->desc)), 200); ?>
-                <a href="<?php echo Route::_($link); ?>" class="new-module mb-3 comModulesSelectCard" data-function="' . $this->escape($function) : ''; ?>" aria-label="<?php echo Text::sprintf('COM_MODULES_SELECT_MODULE', $name); ?>">
+                <a href="<?php echo Route::_($link); ?>" class="new-module mb-3 comModulesSelectCard"
+                    <?php echo !empty($function) ? 'data-function="' . $this->escape($function) . '"': ''; ?>
```
Comment (richard67) [inline administrator/components/com_modules/tmpl/select/default.php]:
```suggestion
                    <?php echo !empty($function) ? 'data-function="' . $this->escape($function) . '"' : ''; ?>
```
This should fix PHPCS failing in drone.

**Comment by chmst** [comment]
Hopefully thats correct now; removed wrong indents

**Comment by Quy** [comment]
Before PR:
38959-heading

After PR (missing heading & button):
38959-new-button

**Comment by chmst** [comment]
Fix toolbar, thank you @Quy

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on e9570276caf00bffdfe5529250e49404fc49a332<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38959">issues.joomla.org/tracker/joomla-cms/38959</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38959">issues.joomla.org/tracker/joomla-cms/38959</a>.</sub>

**Comment by fancyFranci** [comment]
Thank you for cleaning this up!

**Pull request merged**